### DISCUSSION TITLE
Terrain for non-geographic maps

### DISCUSSION THREAD

**Discussion by Sgambe33** [maplibre/maplibre-gl-js#5206] [Q&A]
I'm developing a website to display heatmaps of player movements in a game map. Obviously I am not using geographic coordinates but I managed to load the basemap and display the heatmap.

![image](https://github.com/user-attachments/assets/ff64b7d2-bd26-415b-bb68-87c540a338e5)

 I was wondering if given a png heightmap is possible to apply it directly without tiling. I tried with deck.gl terrain layer but the two thing are not compatible it seems: no maplibre's heatmap over deck.gl's terrain. Is there a solution or I'm better off to keep it 2d? Thanks

**Comment by HarelM** [comment]
The terrain feature in maplibre-gl-js is only supported using raster-dem source, which is tiled, so you'd need to create that kind of source in order to use it with terrain, you could also use it for hill shading, so that's a nice benefit.
There some tools to convert heightmap to terrainRGB tiles out there...

**Reply by Sgambe33** [reply]
So basically I have to implement a way to load a single image instead of a tiled service right?

**Reply by HarelM** [reply]
You might be able to use `addProtocol` for that, load the image to memory and split it on the client side to tiles as needed.
But generally, yes, you'd need to implement it.
